net/isc-dhcp/files/dhcpd.init

```diff
@@ -532,41 +544,25 @@ general_config() {
 			create_empty_zone "$mynet.in-addr.arpa"
 		done
 
-		local need_reload=
-
-		cp -p $conf_local_file ${conf_local_file}_
-
-		cat <<EOF > $conf_local_file
-zone "$domain" {
-	type master;
-	file "$dyndir/db.$domain";
-	update-policy {
-		grant $session_key_name zonesub any;
-	};
-};
-
-EOF
+		rndc addzone $domain "{
+			type primary;
+			file \"$dyndir/db.$domain\";
+			update-policy {
+				grant $session_key_name zonesub any;
+			};
+		};" > /dev/null 2>&1 || true
 
 		for mynet in $rfc1918_nets; do
 			mynet="$(rev_str "$mynet" ".")"
-			cat <<EOF >> $conf_local_file
-zone "$mynet.in-addr.arpa" {
-	type master;
-	file "$dyndir/db.$mynet.in-addr.arpa";
-	update-policy {
-		grant $session_key_name zonesub any;
-	};
-};
-
-EOF
+			rndc addzone $mynet.in-addr.arpa "{
+				type primary;
+				file \"$dyndir/db.$mynet.in-addr.arpa\";
+				update-policy {
+					grant $session_key_name zonesub any;
+				};
+			};" > /dev/null 2>&1 || true
```

Comment:
Ditto

Reply:
If I restart the service, I get:

```
root@OpenWrt:/# /etc/init.d/dhcpd restart
rndc: 'addzone' failed: permission denied
Not allowing new zones in view '_default'
rndc: 'addzone' failed: permission denied
Not allowing new zones in view '_default'
root@OpenWrt:/# 
```

Reply:
> Should `stderr` be piped to `logger`? Or at least not discarded?

The reason I didn't want to do it is that it errors out if the zone already exists (like it will if `isc-dhcp` is restarted). But logging might be a good option...that'll at least show the result from `addzone`

Reply:
> If I restart the service, I get:
> 
> ```
> root@OpenWrt:/# /etc/init.d/dhcpd restart
> rndc: 'addzone' failed: permission denied
> Not allowing new zones in view '_default'
> rndc: 'addzone' failed: permission denied
> Not allowing new zones in view '_default'
> root@OpenWrt:/# 
> ```

Is `bind` configured to allow new zones? (like in the `bind` PR....)

Reply:
What about `modzone`?